src/main/kotlin/app/revanced/patcher/Patcher.kt

```diff
@@ -29,12 +29,13 @@ class Patcher(
         val newDexFile = object : DexFile {
             override fun getClasses(): Set<ClassDef> {
                 // this is a slow workaround for now
-                val mutableClassList = cache.classes.toMutableList()
+                val classes = cache.classes.toMutableSet()
                 cache.classProxy
                     .filter { it.proxyUsed }.forEach { proxy ->
-                        mutableClassList[proxy.originalIndex] = proxy.mutatedClass
+                        classes.remove(classes.elementAt(proxy.originalIndex))
+                        classes.add(proxy.mutatedClass)
                     }
```

Comment:
This looks invalid. Use something like `insert` instead.

Reply:
we first remove the original class from the set of classes, then add the mutable class to the list, what's invalid about it?

Reply:
You remove the original class at index `proxy.originalIndex`, then you add the modified class at index `classes.size`.

Reply:
Yeah that's the intended way

Reply:
Yep, forgot about that. Order doesn't matter here.